### PR TITLE
Mesh: preserve geometry precision in PyVista conversion

### PR DESCRIPTION
**Pull request by peterdsharpe**
# PhysicsNeMo Pull Request

## Description

`from_pyvista()` and `to_pyvista()` previously converted all geometry to `float32`. That can silently destroy valid `float64` geometry when coordinates have a large offset.

At `1e8`, adjacent `float32` values are eight units apart. A one-unit triangle therefore collapses during conversion:

```text
float64 input                            after forced float32

C ●                                      ● A = B = C
  |\
  | \
A ●──● B

area = 0.5                               area = 0

A = (1e8,     1e8)
B = (1e8 + 1, 1e8)
C = (1e8,     1e8 + 1)
```

This PR:

- Preserves real `float32` and `float64` coordinate dtypes during PyVista import and export.
- Retains PhysicsNeMo's existing conversion to `float32` for integer, reduced-precision, and complex geometry.
- Applies the same policy when importing cell centroids.
- Adds regression coverage for large-offset round trips, cell areas, centroid conversion, and explicit `float32` normalization.

Preserving `float64` can increase memory use and keep downstream geometry calculations in double precision. Callers that intentionally want the previous normalization can use `.to(torch.float32)`.

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Dependencies

None.

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

**Comment by peterdsharpe** [comment]
/ok to test https://github.com/NVIDIA/physicsnemo/pull/1781/commits/cb833432a5f091c38d30a70b3be834e28d85e7a5

**Comment by peterdsharpe** [comment]
/ok to test https://github.com/NVIDIA/physicsnemo/pull/1781/commits/d62440d24581ad3e90172997a207fc445ea9cf16

File: physicsnemo/mesh/io/io_pyvista.py [hunk 1]
```diff
@@ -62,6 +62,17 @@ def _tensor_to_vtk_numpy(tensor: torch.Tensor) -> np.ndarray:
...
 
+def _geometry_to_vtk_numpy(tensor: torch.Tensor) -> np.ndarray:
+    """Convert coordinates using PhysicsNeMo's PyVista dtype policy."""
+    tensor = tensor.detach()
+    if tensor.dtype not in (torch.float32, torch.float64):
+        # PyVista/VTK can store some additional coordinate dtypes, but
+        # PhysicsNeMo has historically exported integer, reduced-precision,
+        # and complex geometry as float32. Keep that compatibility policy.
+        tensor = tensor.float()
+    return tensor.cpu().resolve_conj().resolve_neg().numpy()
```
[the remark this answers is not in the document]
Comment (peterdsharpe) [reply physicsnemo/mesh/io/io_pyvista.py:73]: This is not an issue. On the supported PyTorch runtimes, casting complex tensors to a real dtype succeeds and emits PyTorch’s own `UserWarning: Casting complex values to real discards the imaginary part`; it does not raise. The warning originates in ATen rather than this repository, which is why the string is not present in the production source. The focused complex regression passes on PyTorch 2.12, and Coverage CI ran the file successfully on PyTorch 2.11 as part of the 2,465-test passing suite. The `from_pyvista()` path uses the same successful cast behavior.

File: physicsnemo/mesh/io/io_pyvista.py [hunk 2]
```diff
@@ -62,6 +62,17 @@ def _tensor_to_vtk_numpy(tensor: torch.Tensor) -> np.ndarray:
     return tensor.resolve_conj().resolve_neg().numpy()
 
 
+def _geometry_to_vtk_numpy(tensor: torch.Tensor) -> np.ndarray:
+    """Convert coordinates using PhysicsNeMo's PyVista dtype policy."""
+    tensor = tensor.detach()
+    if tensor.dtype not in (torch.float32, torch.float64):
+        # PyVista/VTK can store some additional coordinate dtypes, but
+        # PhysicsNeMo has historically exported integer, reduced-precision,
```
Comment (laserkelvin) [inline physicsnemo/mesh/io/io_pyvista.py:70]: > ...exported integer...

Just so I understand, this is why you're not guarding with `tensor.is_floating_point() and tensor.dtype not in ...`?

**Review by laserkelvin — APPROVED** [review]
Generally LGTM - if it hasn't already been stated somewhere in the docs, maybe would be good to add something like a tip block to document this FP32 casting behavior for PyVista workflows?

File: physicsnemo/mesh/io/io_pyvista.py (hunk 2 above)
Comment (peterdsharpe) [reply physicsnemo/mesh/io/io_pyvista.py:70]: Yeah, the idea is that if you define a `Mesh()` where the points array is integer-dtype, PyVista/VTK is going to cast it to a float anyway - this just makes it explicit.

**Comment by peterdsharpe** [comment]
/ok to test https://github.com/NVIDIA/physicsnemo/pull/1781/commits/69c40dae0973fc1fbfde72854a5e583fbf42cfa6

**Review by megnvidia — APPROVED** [review]
lgtm
